Source/WebKit/WebProcess/Network/WebLoaderStrategy.cpp

```diff
@@ -415,7 +415,8 @@ void WebLoaderStrategy::scheduleLoadFromNetworkProcess(ResourceLoader& resourceL
     bool isMainFrameNavigation = resourceLoader.frame() && resourceLoader.frame()->isMainFrame() && resourceLoader.options().mode == FetchOptions::Mode::Navigate;
     addParametersShared(frame, loadParameters, isMainFrameNavigation);
 
-    loadParameters.serviceWorkersMode = resourceLoader.options().loadedFromOpaqueSource == LoadedFromOpaqueSource::No ? resourceLoader.options().serviceWorkersMode : ServiceWorkersMode::None;
+    auto shouldSkipServiceWorker = resourceLoader.options().loadedFromOpaqueSource == LoadedFromOpaqueSource::Yes && resourceLoader.cachedResource() && resourceLoader.cachedResource()->type() != CachedResource::Type::MediaResource;
```

Comment:
Why are we explicitly allowing media resources to be processed by the service worker?
The rationale should be explained in the commit message.